meson.build

```diff
@@ -0,0 +1,440 @@
+project('libffi', 'c', version : '3.4.4',
+        meson_version : '>= 0.64.0',
+        default_options : ['buildtype=debugoptimized',
+                           'warning_level=1'])
+
+fs = import('fs')
+
+libtool_version = run_command('meson-scripts/extract-libtool-version.py',
+  meson.current_source_dir() / 'libtool-version',
+  check: true).stdout().split(':')
+current = libtool_version[0].to_int()
+revision = libtool_version[1].to_int()
+age = libtool_version[2].to_int()
+ffi_version = '@0@.@1@.@2@'.format(current - age, age, revision)
+ffi_soversion = '@0@'.format(current - age)
+ffi_darwin_versions = '@0@'.format(current + 1)
+
+# NOTE: host = "cross" or "target"
+host_cpu_family = host_machine.cpu_family()
+host_system = host_machine.system()
+
+cc = meson.get_compiler('c')
+is_msvc_like = cc.get_argument_syntax() == 'msvc'
+is_gnu_like = cc.get_argument_syntax() == 'gcc'
+
+if is_gnu_like
+  add_project_arguments('-fexceptions', language: 'c')
+endif
+
+ffi_conf = configuration_data()
+ffi_conf.set_quoted('VERSION', meson.project_version())
+
+size_t = cc.sizeof('size_t')
+ffi_conf.set('SIZEOF_SIZE_T', size_t)
+
+if cc.has_header('stdlib.h') and cc.has_header('string.h')
+  ffi_conf.set('STDC_HEADERS', 1)
+endif
+
+headers = [
+  'sys/memfd.h',
+  'alloca.h',
+  'inttypes.h',
+  'stdint.h',
+  'stdio.h',
+  'stdlib.h',
+  'strings.h',
+  'string.h',
+  'sys/stat.h',
+  'sys/types.h',
+  'unistd.h',
+  'dlfcn.h',
+]
+
+foreach h : headers
+  if cc.has_header(h)
+    define = 'HAVE_' + h.underscorify().to_upper()
+    ffi_conf.set(define, 1)
+  endif
+endforeach
+
+functions = [
+  'memfd_create',
+  'memcpy',
```

Comment:
also is it really necessary to check for stuff like this, or for standard C headers? seems entirely pointless

Reply:
It is only necessary to check for stuff like this, if the source code makes use of the resulting defines. Which... libffi does. HAVE_MEMFD_CREATE is used in src/closures.c, and HAVE_MEMCPY is in ffi_common.h as a polyfill.

Reply:
sure, i'd probably just always define them to 1 :)